bundles/framework/announcements/tool/AnnouncementsTool.js

```diff
@@ -0,0 +1,367 @@
+                template.find('input[name=publisher-announcements]').val(me.annTitles.toString()).attr('announcement-name', me.annTitles.toString());
+            }
+            return template;
+        },
+
+        /**
+         * Creates and opens the dialog from which to choose the announcements.
+         *
+         * @method _openAnnouncementsDialog
+         */
+        _openAnnouncementsDialog: function () {
+            var me = this,
+                popup = Oskari.clazz.create('Oskari.userinterface.component.Popup'),
+                closeButton = Oskari.clazz.create('Oskari.userinterface.component.Button'),
+                title = me.__instance._localization.announcementsTool.popup.title,
+                content = me.templates.announcementsPopup.clone(),
+                announcementInput,
+                annName,
+                i;
+            closeButton.setTitle(me.__instance._localization.announcementsTool.popup.close);
+            closeButton.setHandler(function () {
+                popup.close(true);
+                me.isAnnouncementsDialogOpen = false;
+            });
+            var aLen = this.announcements.length;
+
+            for (i = 0; i < aLen; ++i) {
+                announcementInput = me.templates.inputCheckbox.clone();
+
+                annName = this.announcements[i].title;
+
+                announcementInput.find('input[type=checkbox]').attr({
+                    'id': this.announcements[i].id,
+                    'name': 'announcement',
+                    'value': this.announcements[i].title
+                });
+                announcementInput.find('label').html(annName).attr({
+                    'for': this.announcements[i].title
+                });
+                if (this.selectedAnnouncements.includes(this.announcements[i]) || me.shouldPreselectAnnouncement(this.announcements[i].id)) {
+                    announcementInput.find('input[type=checkbox]').prop('checked', true);
+                }
+
+                content.find('div#publisher-announcements-inputs').append(announcementInput);
+            }
+
+            // WHAT TO DO WHEN ANNOUNCEMENTS ARE SELECTED
+            content.find('input[name=announcement]').on('change', function () {
+                var announcement = me._getItemByCode(jQuery(this).val(), me.announcements);
+                // check if announcement is already checked, if is, add/remove accordingly
+                if (!this.checked) {
+                    me.selectedAnnouncements = me.selectedAnnouncements.filter(function (ann) {
+                        return ann.title !== announcement.title;
+                    });
+                    me.annTitles = me.annTitles.filter(function (e) { return e !== announcement.title; });
+                    me._sendAnnouncementsChangedEvent(me.selectedAnnouncements);
+                } else {
+                    me.selectedAnnouncements.push(announcement);
+                    me.annTitles.push(announcement.title);
```

Comment:
Why would you keep track of this in 2 separate arrays?

Reply:
Just for ease and not needing to filter trough all the announcement data every time we need just the title.

Reply:
I'm not sure what you mean. The other array has the announcements as is while the other has just the titles? Wouldn't it make more sense to have just one array and instead of getting the title with `arrayItem` you would get `arrayItem.title` when you need just the title?